[12.0][ADD] l10n_es_aeat_partner_check: Comprobación recargo equiv.

hildickethan opened the pull request:
Se propone estos cambios para poder comprobar contra la AEAT si la empresa está sometido al recargo de equivalencia e indica la última fecha de comprobación.

Este proceso se hace junto con la petición del resto de la información, pero no se puede utilizar la funcionalidad de `l10n_es_aeat` de `soap` ya que no hay WebService de la información que queremos, estamos haciendo una petición directa al formulario y "parseando" el HTML devuelto para ver el estado de la empresa.

Puse la actualización de fecha en las 2 peticiones por si solo se quiere hacer la petición de recargo (mi caso de uso personal), que se actualize

pedrobaeza commented:
Gracias por la contribución. ¿Podría esto venir como parte de `l10n_es_aeat_partner_check` directamente?

hildickethan commented:
> Gracias por la contribución. ¿Podría esto venir como parte de `l10n_es_aeat_partner_check` directamente?

Si lo crees conveniente, en caso de combinarlos supongo que seria conveniente tenerlo configurable

pedrobaeza commented:
A qué te refieres con configurable? Tal vez convendría tener un campo selection para categorizar el tipo de cliente que es, incluida esta condición?

hildickethan commented:
> A qué te refieres con configurable? Tal vez convendría tener un campo selection para categorizar el tipo de cliente que es, incluida esta condición?

Con configurable me referia a que se haga con configuración para los que ya tienen el módulo instalado, a lo mejor no necesitan/quieren esta comprobación adicional. 
¿Con tipo de cliente que quieres decir? Me parece que un boolean para indicar si está sometido al recargo es suficiente, después a base de eso puedes actualizar la posición fiscal del cliente dependiendo del resultado, pero eso no quería hacerlo aquí porque es un módulo que solo es para informar del estado del cliente en la AEAT

pedrobaeza commented:
Yo creo que no es necesario que sea configurable. Lo del campo selection es si por si mañana surgen más casuísticas fiscales, no llenar de casillas todo. También lo de última vez chequeado creo que debería ser algo general del otro módulo, no de esta opción en concreto.

hildickethan commented:
> Yo creo que no es necesario que sea configurable. Lo del campo selection es si por si mañana surgen más casuísticas fiscales, no llenar de casillas todo. También lo de última vez chequeado creo que debería ser algo general del otro módulo, no de esta opción en concreto.

Vale entiendo que entonces seria en lugar de un Boolean, un Selection con opciones como ["Sometido a Recargo de Equivalencia","No sometido a Recargo de Equivalencia"], y ya si hace falta se puede llenar más en un futuro?

pedrobaeza commented:
Sí, podría contener de valores "Régimen estándar" y "Régimen de recargo de equivalencia". Se me ocurren como mínimo otras casuísticas que podrían venir aquí como criterio de caja, lo de las agencias de viajes, lo de compra de oro, etc.

Commit:
[12.0][IMP] l10n_es_aeat_partner_check: Comprobación regargo equiv.

Añade una petición para comprobar si la empresa está sometido al recargo de equivalencia, indicandolo junto con el resto de la información del pedido

hildickethan commented:
@pedrobaeza Se ha movido al módulo base, con los cambios comentados

pedrobaeza commented:
En v12 no se debe aplicar black.

hildickethan commented:
Fuera commit black

pedrobaeza commented on l10n_es_aeat_partner_check/i18n/es.po:
```diff
@@ -70,11 +66,11 @@ msgid "Data different"
 #, python-format
 msgid "Identificado"
-msgstr "Identificado"
+msgstr ""
```
Por qué borrar traducciones?

hildickethan replied:
Se me habria escapado, lo he vuelto a poner

pedrobaeza commented on l10n_es_aeat_partner_check/i18n/es.po:
```diff
@@ -140,11 +134,41 @@ msgstr "No procesado"
+#: code:addons/l10n_es_aeat_partner_check/models/res_partner.py:15
+#: selection:res.partner,aeat_partner_type:0
+#, python-format
+msgid "Régimen de recargo de equivalencia"
```
En un lado está traducido y en otro no?

hildickethan replied:
Fixed

pedrobaeza commented on l10n_es_aeat_partner_check/models/res_partner.py:
```diff
@@ -56,16 +66,25 @@ def aeat_check_partner(self):
+                "aeat_partner_vat": None,
+                "aeat_partner_name": None,
+                "aeat_partner_check_result": "NO IDENTIFICABLE",
+                "aeat_last_checked": fields.Datetime.now(),
```
Puedes escribir este dato también siempre que se compruebe?

hildickethan replied:
¿A qué te refieres cambiar? Como está ahora mismo escribe este dato en las 2 peticiones independientemente, después de hacer el soap y después de hacer la petición de comprobación de recargo

pedrobaeza replied:
Pues la verdad es que ahora no recuerdo a qué me quería referir, la verdad, así que lo dejamos, y si luego recuerdo, vuelvo a comentar.

pedrobaeza commented:
/ocabot merge major
